src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -188,6 +192,19 @@ class AppInfoBaseData(
       )
     }
 
+  @SuppressWarnings(Array("all")) // async/await
+  def podInstanceStatus(podId: PathId, instance: Instance): Future[Option[PodInstanceStatus]] =
+    async { // linter:ignore UnnecessaryElseBranch
+      val maybePodSpec: Option[PodDefinition] = await(podRepository.getVersion(
```

Comment:
@jdef is correct. There is no way to bulk load in ZK; however, I do think the interface would be better as a bulk load interface so that new stores could implement a more efficient API.